code/game/objects/covers_types/covers_buildable.dm

```diff
@@ -44,7 +45,7 @@
 			return TRUE
 
 /obj/covers/dirt_wall/blocks/incomplete/attackby(obj/item/W as obj, mob/user as mob)
-	if (istype(W, /obj/item/weapon/barrier))
+	if (W.type == /obj/item/weapon/barrier)
```

Comment:
whats wrong with the istype proc?

Reply:
Checks for all types including sandbags and others, we need it to check for the direct type.

Reply:
In short, allows sandbags to build the dirt barricade, which is not correct.

Fixes that and checks directly for THAT type.

Reply:
Did you test that functionality? I would expect byond is stupid and checks only for obj/item/weapon/barrier and no subtypes

Reply:
Yes, it is supposed to only check for weapon/barrier (as that is dirt), and no other.

I tried this in-game and it only accepts that type only.